File: Swift/Utils/Utils/SideMenu/UIViewController+SideMenu.m
#import "UIViewController+SideMenu.h"

@implementation UIViewController (SideMenu)

- (SideMenu *)sideMenuViewController {
    UIViewController *iter = self.parentViewController;
    while (iter) {
        if ([iter isKindOfClass:[SideMenu class]]) {
            return (SideMenu *)iter;
        } else if (iter.parentViewController && iter.parentViewController != iter) {
            iter = iter.parentViewController;
        } else {
            iter = nil;
        }
    }
    return nil;
}

- (IBAction)presentLeftMenuViewController:(id)sender {
    [self.sideMenuViewController presentLeftMenuViewController];
}

- (IBAction)presentRightMenuViewController:(id)sender {
    [self.sideMenuViewController presentRightMenuViewController];
}

@end

File: Swift/Utils/SideMenu/UIViewController+SideMenu.swift
import UIKit

extension UIViewController {
    
    var sideMenuViewController: SideMenu? {
        get {
            var iter: UIViewController? = self.parentViewController
            while (iter != nil) {
                if (iter is SideMenu) {
                    return iter as? SideMenu
                } else if (iter!.parentViewController != nil && iter!.parentViewController != iter) {
                    iter = iter!.parentViewController
                } else {
                    iter = nil;
                }
            }
            return nil;
        }
    }
    
    @IBAction func presentLeftMenuViewController(sender: AnyObject) {
        self.sideMenuViewController!.presentLeftMenuViewController()
    }
    
    @IBAction func presentRightMenuViewController(sender: AnyObject) {
        self.sideMenuViewController!.presentRightMenuViewController()
    }
}
